bindata/network/openshift-sdn/sdn-ovs.yaml

```diff
@@ -34,131 +34,128 @@ spec:
         - |
           #!/bin/bash
           set -euo pipefail
-          chown -R openvswitch:openvswitch /var/run/openvswitch
-          chown -R openvswitch:openvswitch /etc/openvswitch
-
-          # if another process is listening on the cni-server socket, wait until it exits
-          retries=0
-          while true; do
-            if /usr/share/openvswitch/scripts/ovs-ctl status &>/dev/null; then
-              echo "warning: Another process is currently managing OVS, waiting 15s ..." 2>&1
-              sleep 15 & wait
-              (( retries += 1 ))
-            else
-              break
-            fi
-            if [[ "${retries}" -gt 40 ]]; then
-              echo "error: Another process is currently managing OVS, exiting" 2>&1
-              exit 1
+          export SYSTEMD_IGNORE_CHROOT=yes
+          # Check to see if ovs is provided by the node:
+          if systemctl is-enabled ovsdb-server; then
+            echo "openvswitch is running in systemd"
+            # In some very strange corner cases, the owner for /run/openvswitch
+            # can be wrong, so we need to clean up and restart.
+            ovs_uid=$(chroot /host id -u openvswitch)
+            ovs_gid=$(chroot /host id -g openvswitch)
+            chown -R "${ovs_uid}:${ovs_gid}" /run/openvswitch
```

Comment:
Doesn't this just move the problem?

When MCO and CNO have different uid/gid for the openvswitch user/group the CNO got permission denied for things created by MCO.  The result was OVN didn't come up.  Doing `chown -R` to set the uid/gid to what MCO uses doesn't change the fact that the CNO container is using a different uid/gid for the openvswitch user/group.  When a  CNO container tries to access these files using a different uid/gid are you sure that CNO will not hit the same permission denied problem(s)?

Reply:
Everything except openvswitch runs as root. So, yes, we should be fine for permissions. We're also not writing anything to this directory from a container anymore.

Reply:
my understanding here is that we are not trying to set the permissions for this container, we are trying to fix the permissions on the host for the ovs directory because it may have a conflict with kubelet. We had similar problems in openstack between OVS and nova. I guess you could argue that this really isn't this container's job and the problem should be fixed by MCO?

On the other hand while this container will not issue any ovs commands, the ovn-k8s container will. Does the uid/gid matter in that case to just call the binaries?

Reply:
I should have been more specific. If openvswitch is a systemd service, no containerized processes will write to `/run/openvswitch`. They'll only read.

Reply:
There were issues with br-int.mgmt being created but didn't have permission.  Is this OVS or OVN?

Reply:
Where did you see these issues? In CI? Or in your PR?